src/main/java/com/coveo/spillway/storage/AsyncLimitUsageStorage.java

```diff
@@ -78,6 +78,14 @@ public AsyncLimitUsageStorage(LimitUsageStorage wrappedLimitUsageStorage) {
     return wrappedLimitUsageStorage.debugCurrentLimitCounters();
   }
 
+  public void shutdownStorage() {
+    executorService.shutdown();
```

Comment:
Yes you have to do it when you shutdown the app, but I don't see your point of counting the events on the remote storage if you are shutting down everything :)